lib/ASTGen/Sources/ASTGen/Misc.swift

```diff
@@ -26,4 +26,46 @@ extension ASTGenVisitor {
   public func visit(_ node: ArrayElementSyntax) -> ASTNode {
     visit(node.expression)
   }
+
+  /// Form a source location at the given absolute position in the current
+  /// file.
+  public func bridgedSourceLoc(at position: AbsolutePosition) -> BridgedSourceLoc {
+    return createBridgedSourceLoc(at: position, in: base)
+  }
+
+  /// Form a source location at the given node's position in the current file.
+  public func bridgedSourceLoc<Node: SyntaxProtocol>(for node: Node?) -> BridgedSourceLoc {
+    guard let node = node else {
+      return .none
+    }
+    return createBridgedSourceLoc(at: node.position, in: base)
+  }
+}
+
+extension BridgedSourceLoc {
+  static var none: BridgedSourceLoc = BridgedSourceLoc(raw: nil)
+}
+
+extension BridgedIdentifier {
+  static var none: BridgedIdentifier = BridgedIdentifier(raw: nil)
+}
+
+/// Form a source location at the given absolute position in `buffer`.
+func createBridgedSourceLoc(
+  at position: AbsolutePosition,
+  in buffer: UnsafeBufferPointer<UInt8>
+) -> BridgedSourceLoc {
```

Comment:
What do you think about making these conversion functions initializers on the type they return? Feels more swifty to me.

Reply:
When you say "these" are there more than just this one?

Reply:
I was thinking maybe also `bridgedSourceRange`, `bridgedSourceLoc`. Or do they require context that you otherwise need to explicitly pass?

Reply:
They would require passing in `base` in all the calls. Maybe that's okay though.

Reply:
I would need to try these things myself to see what feels better

Reply:
I've changed one to an initializer and kept the others as functions (so there's still bridged functions on the visitor + source manager).